pkg/auth/providers/common/usermanager.go

```diff
@@ -332,3 +342,12 @@ func groupPrincipalPRTB(obj interface{}) ([]string, error) {
 	}
 	return gp, nil
 }
+
+func providerExists(principalIDs []string, provider string) bool {
+	for _, id := range principalIDs {
+		if strings.Contains(id, provider) {
```

Comment:
is it safe to use `contains` here? What are the rules/examples of principalIds vs provider?

Reply:
it would be safer to split on `://` (or just the first `:`) and then do contains